src/statsd_client.cpp

```diff
@@ -50,6 +50,17 @@ StatsdClient::StatsdClient(const string& host, int port, const string& ns)
     config(host, port, ns);
     srandom(time(NULL));
 }
+       
+StatsdClient::StatsdClient(const StatsdClient &original)
+{
+  // Must create our own storage for ClientData object; don't share original's
+  d = new _StatsdClientData;
```

Comment:
This is unsafe. You should not be managing raw pointers in your class.
If you use std::unique_ptr it will solve this problem.
As it stands this class is broken because it does not implement the rule of three/five (ie there is no assignment operator)